accord-core/src/main/java/accord/utils/async/AsyncResults.java

```diff
@@ -104,7 +124,11 @@ protected void start(BiConsumer<? super V, Throwable> callback)
         void setResult(V result, Throwable failure)
         {
             if (!trySetResult(result, failure))
-                throw new IllegalStateException("Result has already been set on " + this);
+            {
+                IllegalStateException f = new IllegalStateException("Result has already been set on " + this);
+                f.addSuppressed(failure);
```

Comment:
I don't understand how we know that `failure` is not null here. It could be and then `Throwable.addSuppressed` will throw NPE. I think we need to check if `failure` is not null before adding it as `suppressed`.

Reply:
good catch, this was true in the seed I hit this on; ill fix